lib/galaxy_test/api/test_genomes.py

```diff
@@ -0,0 +1,24 @@
+from galaxy.exceptions import (
+    MissingDataError,
+    ObjectNotFound,
+)
+from ._framework import ApiTestCase
+
+
+class GenomesApiTestCase(ApiTestCase):
+
+    def test_index(self):
+        response = self._get('genomes')
+        self._assert_status_code_is(response, 200)
+        assert response.json()[0][0] == 'unspecified (?)'
+
+    def test_show_invalid(self):
+        response = self._get('genomes/invalid')
+        self._assert_status_code_is(response, 404)
+        assert response.json()['err_code'] == ObjectNotFound.err_code.code
+
+    def test_show_missing_data(self):
```

Comment:
I'm tempted to request making this an integration test? I don't think it is valid to assume the API tests will be run without this set - we can sort of guarantee this is the case with an integration test though.

While that would be the correct thing to do, I'm worried it is overkill for this API if the API isn't being used. Do you know if this API is being used by the GUI?

Reply:
The `api/genomes` and `api/genomes/{id}` endpoints are used by the client (for trackster and more), so I don't think it would be overkill. Besides, this test fails: it passes locally, so I think the failure is the precisely for the reason you bring up: `anoGam3` doesn't exist on the test instance, so it's raising a different error (i.e., the test isn't testing what is should be testing). I'll write an integration test for this.

Reply:
Sounds good! Thanks for the update.

Reply:
If these integration tests pass remotely, I'll refactor them (wrap each in a context manager, at least). The tests use minimal patching and cover essential functionality only; they ensure that the new error handling works as expected (`MissingDataError` raised when ref data for a valid genome is missing, and `ObjectNotFound` when the provided genome id is invalid), and that data is returned as expected.